File: Demo4/MainActivity.java
package com.example.demo4;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void secondScreenCode(View v)
    {
        Intent intent = new Intent(MainActivity.this, SecondActivity.class);

        intent.putExtra("NAME", "Mithibai");

        startActivity(intent);
    }
}

File: Intent Demo/MainActivity.kt
package com.example.demo4

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    fun secondScreenCode(v: View) {
        val intent = Intent(this, SecondActivity::class.java)
        intent.putExtra("NAME", "XYZ")
        startActivity(intent)
    }
}
